networkx/algorithms/centrality/trophic.py

```diff
@@ -40,6 +40,37 @@ def trophic_levels(G, weight="weight"):
     ----------
     .. [1] Stephen Levine (1980) J. theor. Biol. 83, 195-207
     """
+
+    def bfs_reachable_from_basal(basal_nodes, graph):
+        visited = set()
+        reachable_nodes = set()  # Initialize an empty set to store reachable nodes
+        for node in basal_nodes:
+            queue = [node]
+            while queue:
+                current = queue.pop(0)
+                if current not in visited:
+                    visited.add(current)
+                    reachable_nodes.add(
+                        current
+                    )  # Add the current node to reachable_nodes
+                    for neighbor in graph.successors(current):
+                        if neighbor not in visited:
+                            queue.append(neighbor)
+        return reachable_nodes  # Return the set of reachable nodes
```

Comment:
I think this is already available as:
`[node for layer in nx.bfs_layers(G, sources=basal_nodes) for node in layer]`
Since you just need the number of nodes, you could also use:
`sum(len(layer) for layer in nx.bfs_layers(G, sources=basal_nodes))`

Reply:
I like @dschult's suggestion to compare `sum(len(layer) for layer in nx.bfs_layers(G, sources=basal_nodes))` to `len(G)`, which I expect would be (at least a little) faster.